applications/infection_classification/Infection_classification_25DModel.py

```diff
@@ -0,0 +1,142 @@
+# %%
+import torch
+import lightning.pytorch as pl
+import torch.nn as nn
+
+from pytorch_lightning.loggers import TensorBoardLogger
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+from viscy.transforms import RandWeightedCropd
+from viscy.transforms import NormalizeSampled
+from viscy.data.hcs import HCSDataModule
+from applications.infection_classification.classify_infection_25D import (
+    SemanticSegUNet25D,
+)
+
+from iohub.ngff import open_ome_zarr
+
+# %% Create a dataloader and visualize the batches.
+
+# Set the path to the dataset
+dataset_path = "/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/datasets/Exp_2023_11_08_Opencell_infection/OC43_infection_timelapse_trainVal.zarr"
+
+# find ratio of background, uninfected and infected pixels
+zarr_input = open_ome_zarr(
+    dataset_path,
+    layout="hcs",
+    mode="r+",
+)
+in_chan_names = zarr_input.channel_names
+
+num_pixels_bkg = 0
+num_pixels_uninf = 0
+num_pixels_inf = 0
+num_pixels = 0
+for well_id, well_data in zarr_input.wells():
+    well_name, well_no = well_id.split("/")
+
+    for pos_name, pos_data in well_data.positions():
+        data = pos_data.data
+        T, C, Z, Y, X = data.shape
+        out_data = data.numpy()
+        for time in range(T):
+            Inf_mask = out_data[time, in_chan_names.index("Inf_mask"), ...]
+            # Calculate the number of pixels valued 0, 1, and 2 in 'Inf_mask'
+            num_pixels_bkg = num_pixels_bkg + (Inf_mask == 0).sum()
+            num_pixels_uninf = num_pixels_uninf + (Inf_mask == 1).sum()
+            num_pixels_inf = num_pixels_inf + (Inf_mask == 2).sum()
+            num_pixels = num_pixels + Z * X * Y
+
+pixel_ratio_1 = [
+    num_pixels / num_pixels_bkg,
+    num_pixels / num_pixels_uninf,
+    num_pixels / num_pixels_inf,
+]
+pixel_ratio_sum = sum(pixel_ratio_1)
+pixel_ratio = [ratio / pixel_ratio_sum for ratio in pixel_ratio_1]
```

Comment:
This is repeated in the training scripts and can be abstracted out into a preprocessing function.

Reply:
Can I add this to viscy/utils/image_utils ?

Reply:
I have moved it to preprocessing as a file.